src/SFA.DAS.ApprenticePortal.OuterApi.Mock/Models/Apprentice.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Newtonsoft.Json;
+using System;
+
+namespace SFA.DAS.ApprenticePortal.OuterApi.Mock.Models
+{
+    public sealed class Apprentice
+    {
```

Comment:
My only concern is we are duplicating the model in the application here. Would it work as well if we referenced the models directly from the application layer and have the With methods work as extension methods?    

In some other testing projects, we mix unit testing with 'the Specflow testing, and therefore will have issues resolving which Apprentice model to use

Reply:
This `Apprentice` class has an extra data property that the domain model doesn't.  I think that in a lot of projects this won't be the case and it would be possible to use the domain model and put the `With` methods as extensions.